code/datums/gamemodes/distress.dm

```diff
@@ -1,85 +1,90 @@
 	flags_round_type = MODE_INFESTATION|MODE_LZ_SHUTTERS|MODE_XENO_RULER
 	flags_landmarks = MODE_LANDMARK_SPAWN_XENO_TUNNELS|MODE_LANDMARK_SPAWN_MAP_ITEM
-
 	round_end_states = list(MODE_INFESTATION_X_MAJOR, MODE_INFESTATION_M_MAJOR, MODE_INFESTATION_X_MINOR, MODE_INFESTATION_M_MINOR, MODE_INFESTATION_DRAW_DEATH)
 
-	var/round_stage = DISTRESS_MARINE_DEPLOYMENT
-
-	var/list/survivors = list()
+	valid_job_types = list(
+		/datum/job/terragov/command/captain = 1,
+		/datum/job/terragov/command/fieldcommander = 1,
+		/datum/job/terragov/command/staffofficer = 4,
+		/datum/job/terragov/command/pilot = 2,
+		/datum/job/terragov/police/chief = 1,
+		/datum/job/terragov/police/officer = 5,
+		/datum/job/terragov/engineering/chief = 1,
+		/datum/job/terragov/engineering/tech = 1,
+		/datum/job/terragov/requisitions/officer = 1,
+		/datum/job/terragov/medical/professor = 1,
+		/datum/job/terragov/medical/medicalofficer = 6,
+		/datum/job/terragov/medical/researcher = 2,
+		/datum/job/terragov/civilian/liaison = 1,
+		/datum/job/terragov/silicon/synthetic = 1,
+		/datum/job/terragov/silicon/ai = 1,
+		/datum/job/terragov/squad/engineer = 8,
+		/datum/job/terragov/squad/corpsman = 8,
+		/datum/job/terragov/squad/smartgunner = 1,
+		/datum/job/terragov/squad/specialist = 1,
+		/datum/job/terragov/squad/leader = 1,
+		/datum/job/terragov/squad/standard = -1,
+		/datum/job/survivor/rambo = 1,
+		/datum/job/xenomorph = 2,
+		/datum/job/xenomorph/queen = 1
+	)
 
-	var/surv_starting_num 	  = 0
-	var/marine_starting_num   = 0
+	var/round_stage = DISTRESS_MARINE_DEPLOYMENT
 
 	var/bioscan_current_interval = 45 MINUTES
 	var/bioscan_ongoing_interval = 20 MINUTES
-
-	latejoin_larvapoints		= 0
-	latejoin_larvapoints_required = 9 //in case config doesn't deliver a value in initialize_scales() for some reason
 	var/orphan_hive_timer
 
 
-/datum/game_mode/distress/announce()
+/datum/game_mode/infestation/distress/announce()
 	to_chat(world, "<span class='round_header'>The current map is - [SSmapping.configs[GROUND_MAP].map_name]!</span>")
 
 
-/datum/game_mode/distress/can_start(bypass_checks = FALSE)
+/datum/game_mode/infestation/distress/can_start(bypass_checks = FALSE)
 	. = ..()
 	if(!.)
 		return
-	var/found_queen = initialize_xeno_leader()
-	var/found_xenos = initialize_xenomorphs()
-	if(!found_queen && !found_xenos && !bypass_checks)
+	var/xeno_candidate = FALSE //Let's guarantee there's at least one xeno.
+	for(var/level in list(JOBS_PRIORITY_HIGH, JOBS_PRIORITY_MEDIUM, JOBS_PRIORITY_LOW))
+		for(var/p in GLOB.ready_players)
+			var/mob/new_player/player = p
+			if(player.client.prefs.job_preferences[/datum/job/xenomorph/queen] == level && SSjob.AssignRole(player, SSjob.GetJobType(/datum/job/xenomorph/queen)))
+				xeno_candidate = TRUE
+				break
+			if(player.client.prefs.job_preferences[/datum/job/xenomorph] == level && SSjob.AssignRole(player, SSjob.GetJobType(/datum/job/xenomorph)))
+				xeno_candidate = TRUE
+				break
+	if(!xeno_candidate && !bypass_checks)
```

Comment:
Might be useful to make this generic, `./proc/ensure_job_selected(/datum/path, number_of_ready_players)`

Reply:
I'd rather not future-proof as much. It's hard to foresee another game mode that will require something like this that's not already part of the `infestation` path.